File: facturacion-backend/src/main/java/edu/yavirac/facturacionbackend/service/ClienteService.java
package edu.yavirac.facturacionbackend.service;

import java.util.List;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import edu.yavirac.facturacionbackend.entity.Cliente;
import edu.yavirac.facturacionbackend.repository.ClienteRepository;

@Service
public class ClienteService {

    @Autowired
    ClienteRepository clienteRepository;

    //Create y Update
    public Cliente save(Cliente cliente)
    {
      return clienteRepository.save(cliente);
    }
    //Read
    public Cliente findById(Long id)
    {
        return clienteRepository.findById(id).get();
    }

    //Delete
    public void deleteById(Long id)
    {
        clienteRepository.deleteById(id);
    }

    public List<Cliente> findAll()
    {
        return clienteRepository.findAll();
    }
    
}

File: facturacion-frontend/src/app/cliente/cliente.service.ts
import { Injectable } from '@angular/core';
import { Cliente } from './cliente';
import { Observable } from 'rxjs';
import { HttpClient, HttpHeaders } from '@angular/common/http';
import { environment } from 'src/environments/environment';

@Injectable({
  providedIn: 'root'
})
export class ClienteService {

  private httpOptions =  {
    headers: new HttpHeaders({"Content-Type":"application/json"})
  }

  private url = environment.API_URL + "/cliente";

  constructor(
    private http: HttpClient
  ) { }

  //Create
  public save(cliente: Cliente): Observable<Cliente>
  {
    return this.http.post<Cliente>
    (this.url+"/save",cliente,this.httpOptions);
  }

  //Read
  public findById(id: number): Observable<Cliente>
  {
    return this.http.get<Cliente>(this.url+"/id", this.httpOptions);
  }

  //Update
  public update(cliente: Cliente): Observable<Cliente>
  {
    return this.http.put<Cliente>(this.url+"/update", cliente, this.httpOptions);
  }

  //Delete
  public deleteById(id: number): Observable<Cliente>
  {
    return this.http.delete<Cliente>(this.url+"/deleteById/"+id, this.httpOptions);
  }

  //(FindAll)
  public findAll(): Observable<Cliente[]>
  {
    return this.http.get<Cliente[]>(this.url+"/findAll", this.httpOptions);
  }

  public findAllByName(termino: string): Observable<Cliente[]>
  {
    return this.http.get<Cliente[]>(this.url+"/findByName/"+termino, this.httpOptions);
  }

}
